### PR TITLE
added devise

### LINKED ISSUE #10: As a new user, I should be able to create an account and login.

### PR DESCRIPTION
**Pull request by ryang72**
## Changes
- added `gem 'devise'`
- `bundle install`
- `rails generate devise:install`
- added default url option in `config/enviroments/development.rb` line 63: `config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }`
- `rails generate devise User` (to create the user model with devise)
- `rake db:migrate` (to commit the user model)
- `rails generate devise:views` (so we can edit the devise login view components)
basically https://riptutorial.com/ruby-on-rails/example/5843/authentication-using-devise

#### Attached issue?
resolves #10

### CODE REVIEW

File: config/initializers/devise.rb [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+
+# Use this hook to configure devise mailer, warden hooks and so forth.
+# Many of these configuration options can be set straight in your model.
+Devise.setup do |config|
+  # The secret key used by Devise. Devise uses this key to generate
+  # random tokens. Changing this key will render invalid all existing
+  # confirmation, reset password and unlock tokens in the database.
+  # Devise will use the `secret_key_base` as its `secret_key`
+  # by default. You can change it below and use your own secret key.
+  # config.secret_key = '666c38a4e620670a693b007038508a3536ff6101142469fbc0311b7ac7c7bc697b01c4ec60e42b0f16ce13d276595c46173817ef6341c4c70197e0af427d9b22'
```
Comment (zhaoalex) [inline config/initializers/devise.rb]: Maybe don't commit this? instead put it in a secrets file and use env

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+  #
+  # Limiting the stretches to just one in testing will increase the performance of
+  # your test suite dramatically. However, it is STRONGLY RECOMMENDED to not use
+  # a value less than 10 in other environments. Note that, for bcrypt (the default
+  # algorithm), the cost increases exponentially with the number of stretches (e.g.
+  # a value of 20 is already extremely slow: approx. 60 seconds for 1 calculation).
+  config.stretches = Rails.env.test? ? 1 : 11
+
+  # Set up a pepper to generate the hashed password.
+  # config.pepper = '92e93b96f84b0a7c25b6e2d1f0f6ddfce686079264eecf6e841789a9d918112813772fb40d0b5045c229c7b4f30af16da61d8925373df58522791f83d2c754ef'
```
Comment (zhaoalex) [inline config/initializers/devise.rb]: maybe same idk

File: db/migrate/20191022191059_devise_create_users.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      t.datetime :remember_created_at
+
+      ## Trackable
+      # t.integer  :sign_in_count, default: 0, null: false
+      # t.datetime :current_sign_in_at
+      # t.datetime :last_sign_in_at
+      # t.inet     :current_sign_in_ip
+      # t.inet     :last_sign_in_ip
+
+      ## Confirmable
```
Comment (zhaoalex) [inline db/migrate/20191022191059_devise_create_users.rb]: Just making sure we aren't going to use confirmation tokens?

File: config/initializers/devise.rb [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+
+  # ==> Controller configuration
+  # Configure the parent class to the devise controllers.
+  # config.parent_controller = 'DeviseController'
+
+  # ==> Mailer Configuration
+  # Configure the e-mail address which will be shown in Devise::Mailer,
+  # note that it will be overwritten if you use your own mailer class
+  # with default "from" parameter.
+  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
```
Comment (zhaoalex) [inline config/initializers/devise.rb]: also please change the `please-change-me` hehe

File: config/initializers/devise.rb (hunk 1 above)
Comment (ryang72) [reply config/initializers/devise.rb]: now dealt with

File: config/initializers/devise.rb (hunk 2 above)
Comment (ryang72) [reply config/initializers/devise.rb]: now dealt with

File: db/migrate/20191022191059_devise_create_users.rb (hunk 3 above)
Comment (ryang72) [reply db/migrate/20191022191059_devise_create_users.rb]: didn't set this up yet, but shouldn't be hard in the future

File: config/initializers/devise.rb (hunk 4 above)
Comment (ryang72) [reply config/initializers/devise.rb]: dealt with

File: app/views/react_hello/index.html.erb [hunk 5]
```diff
@@ -1 +1,9 @@
-<%= react_component 'DummyHello', name: 'React' %>
\ No newline at end of file
+<%= react_component 'DummyHello', name: 'React' %>
+<% if user_signed_in? %>
+  <div>Signed in as... <%= current_user.email %></div>
+  <li><%= link_to 'Edit account', edit_user_registration_path %></li>
+  <li><%= link_to 'Sign out', destroy_user_session_path, :method=>'delete' %></li>
+<% end %>
+<% if not user_signed_in? %>
```
Comment (zhaoalex) [inline app/views/react_hello/index.html.erb]: does `else` not work here LOL
